File: 백준/Gold/1253. 좋다/좋다.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func readLine(reader *bufio.Reader) string {
	line, _ := reader.ReadString('\n')
	line = strings.TrimSpace(line)
	return line
}

func readInt(reader *bufio.Reader) int {
	num, _ := strconv.Atoi(readLine(reader))
	return num
}

func readIntArray(reader *bufio.Reader) []int {
	fields := strings.Fields(readLine(reader))
	numbers := make([]int, len(fields))
	for i, v := range fields {
		num, _ := strconv.Atoi(v)
		numbers[i] = num
	}
	return numbers
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	N := readInt(reader)
	numbers := readIntArray(reader)

	sort.Ints(numbers)

	count := 0

	for i := 0; i < N; i++ {
		target := numbers[i]
		left := 0
		right := N - 1

		for left < right {
			if i == left {
				left++
				continue
			}
			if i == right {
				right--
				continue
			}
			sum := numbers[left] + numbers[right]
			if sum < target {
				left++
			} else if sum > target {
				right--
			} else {
				count++
				break
			}
		}
	}

	fmt.Fprint(writer, count)
}

File: 백준/Gold/1253. 좋다/좋다.py
import sys

input = sys.stdin.readline

number_nr = int(input())
numbers = list(map(int, input().split()))

numbers.sort()

good_num_nr = 0

for target_i in range(0, number_nr):
    target_v = numbers[target_i]
    start_i = 0
    end_i = number_nr - 1
    while start_i < end_i:
        start_v = numbers[start_i]
        end_v = numbers[end_i]
        sum = start_v + end_v
        if start_i == target_i or sum < target_v:
            start_i += 1
            continue
        elif end_i == target_i or sum > target_v:
            end_i -= 1
            continue
        else:
            good_num_nr += 1
            break

print(good_num_nr)
